### PR TITLE
[NO-TICKET] Add on dark stories for typography

### PR DESCRIPTION
**Pull request by line47**
## Summary

- Added on dark stories for all headers, body text, links and lists
- Updated doc site pages for headers, body text, links and lists to add links to github code and Sketch assets 
- Added all headings, all lists and all body text stories 

## How to test

- run `yarn start` and visit the doc site pages for headers, body text, links and lists and see that there are links to Sketch, Storybook and Github for each page
- run `yarn storybook` and see that there are stories for All headings, all headings on dark, links on dark, all body text, all body text on dark, all lists and all lists on dark 

## Checklist

- [x] Prefixed the PR title with the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) as `[WNMGDS-####] Title`

### If this is a change to code:

- [x] Created or updated unit tests to cover the logic added
- [x] If necessary, updated unit-test snapshots (`yarn test:unit:update`) and browser-test snapshots (`yarn test:browser:update`)

### If this is a change to documentation:

- [x] Checked for spelling and grammatical errors

### CODE REVIEW

File: packages/design-system/src/components/List/List.stories.jsx [hunk 1]
```diff
@@ -20,17 +20,51 @@ const listMarkup = (type) => {
...
   return (
     <>
-      <h2 className="ds-h6" id={`${type}-list-id`}>
-        {label} list title
-      </h2>
       <Tag className={className} aria-labelledby={`${type}-list-id`}>
-        <li>List item 1</li>
-        <li>List item 2</li>
+        <li>{type} list item 1</li>
+        <li>{type} list item 2</li>
```
Comment (pwolfert) [inline packages/design-system/src/components/List/List.stories.jsx:25]: Good idea 👍

Comment (pwolfert) [reply packages/design-system/src/components/List/List.stories.jsx:25]: Looks like the `label` variable is now unused and that that could work here
```suggestion
        <li>{label} list item 1</li>
        <li>{label} list item 2</li>
```

File: packages/design-system/src/components/List/List.stories.jsx [hunk 2]
```diff
@@ -20,17 +20,51 @@ const listMarkup = (type) => {
...
+    <ol className="ds-c-list" aria-labelledby="ordered-list-id">
+      <li>ordered list item 1</li>
+      <li>ordered list item 2</li>
+    </ol>
+    <ul className="ds-c-list ds-c-list--bare" aria-labelledby="unstyled-list-id">
+      <li>unstyled list item 1</li>
+      <li>unstyled list item 2</li>
+    </ul>
+  </>
+);
```
Comment (pwolfert) [inline packages/design-system/src/components/List/List.stories.jsx:46]: The function above can be reused here

```suggestion
export const AllLists = () => (
  <>
    {listMarkup("unordered")}
    {listMarkup("ordered")}
    {listMarkup("unstyled")}
  </>
);
```

File: packages/design-system/src/components/List/List.stories.jsx [hunk 3]
```diff
@@ -20,17 +20,51 @@ const listMarkup = (type) => {
...
```
Comment (pwolfert) [inline packages/design-system/src/components/List/List.stories.jsx:63]: We can reuse the whole story above here

```suggestion
export const AllListsOnDark = AllLists.bind({});
```

File: packages/design-system/src/components/utilities/typographyBodyText.stories.tsx [hunk 4]
```diff
@@ -27,6 +27,24 @@ export const AllBodyText = () => {
...
+      </p>
+      <p className="ds-text-body--md">
+        <strong>Medium:</strong> {body}
+      </p>
+      <p className="ds-text-body--sm">
+        <strong>Small:</strong> {body}
+      </p>
+    </>
+  );
+};
```
Comment (pwolfert) [inline packages/design-system/src/components/utilities/typographyBodyText.stories.tsx:44]: Same thing here where the body of the story doesn't change, so we can actually just make a copy of the story this way:

```suggestion
export const AllBodyTextOnDark = AllBodyText.bind({});
```

The `.bind({})` is a bit ugly, but it's what they do in Storybook docs, I'm pretty sure. It's JavaScript for _clone this function, please_.

File: packages/design-system/src/components/utilities/typographyHeadings.stories.tsx [hunk 5]
```diff
@@ -25,6 +25,23 @@ export const AllHeadings = () => {
     </>
   );
 };
+export const AllHeadingsOnDark = () => {
```
Comment (pwolfert) [inline packages/design-system/src/components/utilities/typographyHeadings.stories.tsx:28]: Same thing can be done to this function

File: packages/docs/content/foundation/typography/list.mdx [hunk 6]
```diff
@@ -41,12 +46,6 @@ import loremIpsumGenerator from '../../../src/helpers/loremIpsumGenerator';
     </ol>
 </EmbeddedExample>
 
-<Alert variation="warn" weight="lightweight" class="ds-u-margin-y--3">
-  <p class="ds-u-margin-top--0">It's worth mentioning that when <code>list-style</code> is removed from a list, <a href="https://developer.mozilla.org/en-US/docs/Web/CSS/list-style#accessibility_concerns" target="_blank">Safari will not recognize it as a list</a> in the accessibility tree.</p>
-
-  <p class="ds-u-margin-bottom--0">To resolve this issue, <code>role="list"</code> should be added to a list where the styles have been removed.</p>
-</Alert>
```
Comment (pwolfert) [inline packages/docs/content/foundation/typography/list.mdx:48]: Why was this removed?

Comment (pwolfert) [reply packages/docs/content/foundation/typography/list.mdx:48]: It was added in https://github.com/CMSgov/design-system/pull/2009

Comment (line47) [reply packages/docs/content/foundation/typography/list.mdx:48]: We have it down under the accessibility guidance section already. 
"When using an unstyled list, `role="list"` should be added to the list element." 

I'm also still thinking about if we should keep our examples super clean and free from extra guidance and utilize the guidance section for that purpose?  

Maybe the alert moves down to under the a11y section?

File: packages/design-system/src/components/utilities/typographyHeadings.stories.tsx (hunk 5 above)
Comment (line47) [reply packages/design-system/src/components/utilities/typographyHeadings.stories.tsx]: Thanks for pointing out and suggesting all these great updates @pwolfert :partying\_face:

File: packages/docs/content/foundation/typography/list.mdx (hunk 6 above)
Comment (pwolfert) [reply packages/docs/content/foundation/typography/list.mdx:48]: Didn't realize it was duplicate info 👍

Comment (pwolfert) [reply packages/docs/content/foundation/typography/list.mdx:48]: Maybe we should add this part to the guidance below for a little explanation
```
<a href="https://developer.mozilla.org/en-US/docs/Web/CSS/list-style#accessibility_concerns" target="_blank">Safari will not recognize it as a list</a> in the accessibility tree.</p>
```

File: packages/docs/content/foundation/typography/body.mdx [hunk 7]
```diff
@@ -5,7 +5,7 @@ intro: Body text is used for the majority of the content on the website or appli
 core:
   githubLink: design-system/src/styles/base/_typography.scss
   sketchLink: WKOKl08
-  storybookLink: foundations-typography-bodytext--all-body-text
+  storybookLink: foundations-typography-body-text--all-body-text
```
Comment (zarahzachz) [inline packages/docs/content/foundation/typography/body.mdx:8]: Navigating to Storybook from this page errored out for me. Should probably be `foundations-typography--body`

File: packages/docs/content/foundation/typography/headings.mdx [hunk 8]
```diff
@@ -131,7 +131,7 @@ Heading--4xl is used for the heading on landing pages that serve as an entry poi
 
 
 <ResponsiveExample
-  storyId="foundations-typography--responsive-headings"
+  storyId="foundations-typography-headings--responsive-headings"
```
Comment (zarahzachz) [inline packages/docs/content/foundation/typography/headings.mdx:134]: I'm not seeing Storybook examples embedded here.

File: packages/docs/content/foundation/typography/list.mdx [hunk 9]
```diff
@@ -2,7 +2,14 @@
 title: List
 order: 40
 intro: Lists organize written information for users sequentially.
-
+core:
```
Comment (zarahzachz) [inline packages/docs/content/foundation/typography/list.mdx:5]: I couldn't find List in our docs navigation?

Comment (line47) [reply packages/docs/content/foundation/typography/list.mdx:5]: https://design.cms.gov/foundation/typography/list/?theme=core

File: packages/docs/content/foundation/typography/body.mdx (hunk 7 above)
Comment (line47) [reply packages/docs/content/foundation/typography/body.mdx:8]: This is what I see when I run storybook locally. Maybe I'm not understanding something?
Screen Shot 2022-10-28 at 4 57 10 PM

File: packages/docs/content/foundation/typography/headings.mdx (hunk 8 above)
Comment (line47) [reply packages/docs/content/foundation/typography/headings.mdx:134]: Hmmm, I'm not able to reproduce. This is from me running locally 🤷 
Screen Shot 2022-10-28 at 4 58 45 PM

Comment (pwolfert) [reply packages/docs/content/foundation/typography/headings.mdx:134]: @zarahzachz, could you have forgotten to `yarn build:storybook:docs`?

**Review by pwolfert — APPROVED** [review]
LGTM, except the visual regression images caught an issue with one of the medicare overrides. [See this image](https://github.com/CMSgov/design-system/blob/90b76011025992d642a054d038c96d5dcfd83201/tests/browser/story-screenshots.test.ts-snapshots/foundations-typography-body-text--all-body-text-on-dark-medicare-Mobile-Chrome-linux.png). The override doesn't take into account the inverse class. We'll need to follow up to fix that bug.

File: packages/docs/content/foundation/typography/headings.mdx (hunk 8 above)
Comment (zarahzachz) [reply packages/docs/content/foundation/typography/headings.mdx:134]: Yeah, that's exactly what happened. Disregard my feedback!
